Fix CpiGuard typo, but actually DRY processor methods

CriesofCarrots opened the pull request:
I was reading #3756 to understand what the new extension does, and stumbled on a typo.
I also noticed an opportunity to DRY the processor code, and then realized I introduced the verbose pattern myself in RequiredMemoTransfers. Wdyt about this tidying?

2501babe commented:
100% in favor, i copied the existing structure just because i thought it was done that way on purpose for style reasons

2501babe commented on token/program-2022/src/extension/cpi_guard/processor.rs:
```diff
@@ -88,11 +63,11 @@ pub(crate) fn process_instruction(
         }
         CpiGuardInstruction::Disable => {
             msg!("CpiGuardInstruction::Disable");
-            process_diasble_cpi_guard(program_id, accounts)
```
also thank you for catching this!

CriesofCarrots commented:
> i thought it was done that way on purpose for style reasons

I figured! Originally, only `enable` initialized the extension, so it seemed clearer for the methods to be separate. When I updated the code to also initialize on `disable`, I neglected to clean up after myself (to wit: that incorrect code comment that you caught and fixed)
